pinot-core/src/main/java/org/apache/pinot/core/segment/processing/framework/SegmentProcessorFramework.java

```diff
@@ -91,33 +92,51 @@ public SegmentProcessorFramework(List<RecordReader> recordReaders, SegmentProces
    */
   public List<File> process()
       throws Exception {
+    try {
+      return doProcess();
+    } catch (Exception e) {
+      if (_partitionToFileManagerMap != null) {
+        for (GenericRowFileManager fileManager : _partitionToFileManagerMap.values()) {
+          fileManager.cleanUp();
+        }
+      }
+      throw new RuntimeException("Failed to complete process", e);
```

Comment:
Why changing the exception? you should be able to do `throw e;`

Reply:
Same in `SegmentMapper`

Reply:
I was thinking to add a bit more context before bubbling it up

Reply:
I don't know. I would prefer to get the specific exception why that fail instead of a generic runtime exception and having to check the cause. But it is not a blocker, if you think that it is better to throw the runtime, it is ok to me.

Reply:
updated to throw e directly, as the extra context msg passed to RuntimeException didn't carry much info anway.